### PR TITLE
add support for true in x-b3-sampled

### LINKED ISSUE #1299: Accept `X-B3-Sampled` header to equal `true`
# Feature Request

Right now when you use the `B3Propagator` for the AspNet/AspNetCore instrumentations, the incoming request needs to set the `X-B3-Sampled` header to `1` for the request to be sampled. As mentioned in the [open zipkin repo](https://github.com/openzipkin/b3-propagation#sampling-state-1) some tracer implementations may have set this header to the value `true` instead of `1`. 

Can the B3Propagator be made to be lenient and accept this value as well? This would put much less burden on anyone moving to the open-telemetry instrumentation when services may have this older implementation.

### PR DESCRIPTION
**Pull request by christopher-taormina-zocdoc**
Fixes #1299 

## Changes

Adds a string for the possible `X-B3-Sampled` value of `true` and checks the received header against both values. Setting the header for outgoing requests remains unchanged.

For significant contributions please make sure you have completed the following items:

* [ ] `CHANGELOG.md` updated for non-trivial changes
* [ ] Design discussion issue #
* [ ] Changes in public API reviewed

### CODE REVIEW

File: test/OpenTelemetry.Tests/Trace/Propagation/B3PropagatorTest.cs
Comment (eddynaka) [inline test/OpenTelemetry.Tests/Trace/Propagation/B3PropagatorTest.cs:107]: We could change from Fact to Theory, and pass two inlineData, one using "1" and other using "true".

File: test/OpenTelemetry.Tests/Trace/Propagation/B3PropagatorTest.cs
Comment (eddynaka) [inline test/OpenTelemetry.Tests/Trace/Propagation/B3PropagatorTest.cs:118]: Change from Fact to Theory, adding "0", "false", "other value"

File: src/OpenTelemetry.Api/Context/Propagation/B3Propagator.cs
Comment (eddynaka) [inline src/OpenTelemetry.Api/Context/Propagation/B3Propagator.cs:54]: Should we add in the hashset constructor something like:
new HashSet<string>(StringComparer.OrdinalIgnoreCase)

What do you think?

**Review by cijothomas — APPROVED** [review]

**Comment by cijothomas** [comment]
@chris-smith-zocdoc Can you add a line to the changelog as well? Thank you!

**Review by eddynaka — APPROVED** [review]

File: src/OpenTelemetry.Api/CHANGELOG.md
Comment (cijothomas) [inline src/OpenTelemetry.Api/CHANGELOG.md:1]: some encoding got accidentaly changed to this file? there must have been some CI which detects this.

Comment (christopher-taormina-zocdoc) [reply src/OpenTelemetry.Api/CHANGELOG.md:1]: didn't see that coming, let me take a look

Comment (eddynaka) [reply src/OpenTelemetry.Api/CHANGELOG.md:1]: check if u are using utf8. I saw that sometimes it changed from utf8 to utf8-BOM.

**Review by reyang — APPROVED** [review]
LGTM after the CRLF line ending got removed.

File: src/OpenTelemetry.Api/CHANGELOG.md
Comment (christopher-taormina-zocdoc) [reply src/OpenTelemetry.Api/CHANGELOG.md:1]: I edited it in vs code and it claimed it saved as utf-8

Comment (eddynaka) [reply src/OpenTelemetry.Api/CHANGELOG.md]: you solved :) what was the issue in the end?

Comment (christopher-taormina-zocdoc) [reply src/OpenTelemetry.Api/CHANGELOG.md]: definitely line endings lol

**Comment by cijothomas** [comment]
@christopher-taormina-zocdoc can you merge from the latest master? Usually maintainers/approvers have permission to do this in PR, but this one doesn't seem to have it, so you have to do it.

**Comment by christopher-taormina-zocdoc** [comment]
@cijothomas just merged in master. The commit says `origin-master` but that just the name the branch has on my local git

**Comment by cijothomas** [comment]
Thanks @christopher-taormina-zocdoc

**Pull request merged**
